types/react/index.d.ts

```diff
@@ -3119,9 +3119,9 @@ type ReactManagedAttributes<C, P> = C extends { propTypes: infer T; defaultProps
 
 declare global {
     namespace JSX {
-        interface Element extends React.ReactElement<any, any> { }
+        type Element = React.ReactNode;
```

Comment:
This is the core of the change. It's a hack but it trades the linked issues with new ones (credit to @sebmarkbage).

Reply:
I think the main issue is things like:

```
function Foo() : React.ReactElement {
  return <div />;
}
```

It's not wrong to type it this way since this function will always return an actual element but it would error in a lot of places.